src/tests/unit/tests/reports/assessment-json-export-json-builder.test.ts

```diff
@@ -0,0 +1,493 @@
+        assessmentDataStub.incompleteDetailsData = incompleteDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        expect(generatedJson.results.length).toBe(2);
+    });
+
+    test('item previously marked passed marks incomplete upon creation of incomplete results item', () => {
+        assessmentDataStub.passedDetailsData = passedDetailsDataStub;
+        assessmentDataStub.incompleteDetailsData = incompleteDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        expect(generatedJson.results.find(item => item.wcagNumber === '1.3.1').status).toBe(
+            'incomplete',
+        );
+    });
+
+    test('failed details data adds to set and breaks into WCAG Categories', () => {
+        assessmentDataStub.failedDetailsData = failedDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        expect(generatedJson.results.length).toBe(3);
+    });
+
+    test('failed details data instances properly populate', () => {
+        assessmentDataStub.failedDetailsData = failedDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        expect(
+            generatedJson.results
+                .find(item => item.wcagNumber === '1.3.1')
+                .requirementsFailed.find(item => item.requirementKey === 'label')
+                .instances.find(item => item.path === 'input[name="email"]').snippet,
+        ).toBe('<input type="text" name="email">');
+    });
+
+    test('failed status overrides incomplete and passed statuses', () => {
+        assessmentDataStub.passedDetailsData = passedDetailsDataStub;
+        assessmentDataStub.incompleteDetailsData = incompleteDetailsDataStub;
+        assessmentDataStub.failedDetailsData = failedDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        expect(generatedJson.results.find(item => item.wcagNumber === '1.3.1').status).toBe('fail');
+        expect(generatedJson.results.find(item => item.wcagNumber === '4.1.2').status).toBe('fail');
+    });
+
+    test('data sorts properly after populating', () => {
+        assessmentDataStub.passedDetailsData = passedDetailsDataStub;
+        assessmentDataStub.incompleteDetailsData = incompleteDetailsDataStub;
+        assessmentDataStub.failedDetailsData = failedDetailsDataStub;
+        const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);
+        let wcagNumberForCompare: string = '0.0.0';
+        generatedJson.results.forEach(item => {
+            expect(item.wcagNumber.localeCompare(wcagNumberForCompare)).toBe(1);
+            wcagNumberForCompare = item.wcagNumber;
+        });
+    });
+});
```

Comment:
I realize this would overlap with the tests you're already got, but what would your thoughts be on adding a snapshot test for the whole json output? Like:
```ts
test('matches snapshot', () => {
    assessmentDataStub.passedDetailsData = passedDetailsDataStub;
    assessmentDataStub.incompleteDetailsData = incompleteDetailsDataStub;
    assessmentDataStub.failedDetailsData = failedDetailsDataStub;

    const generatedJson = buildAssessmentJsonExportData(comment, version, assessmentDataStub);

    const jsonString = JSON.stringify(generatedJson, null, '  ');
    expect(jsonString).toMatchSnapshot();
});
```
The main purpose of this would be to create a snapshot with a human-readable output. It could also be done without using snapshots by storing a test json file against which to compare.

This is definitely an optional suggestion! I just found myself wanting to see the actual json output while reviewing the PR.

Reply:
I definitely don't mind the idea of a snapshot, I've added it!